File: benchmarks/array-100K/main.kt
import kotlin.time.measureTime

fun main() {
    val time = measureTime {
        val arr = arrayOfNulls<Int>(100_000)
        for (i in 0..<100_000) {
            arr[i] = i
        }
    }

    println(time.inWholeNanoseconds)
}

File: benchmarks/array-100K/main.zig
const std = @import("std");

pub fn main() !void {
    const before: i128 = std.time.nanoTimestamp();

    var arr: [100000]u32 = undefined;
    for (0..arr.len) |i| {
        arr[i] = @intCast(i);
    }

    const after: i128 = std.time.nanoTimestamp();

    std.debug.print("{}", .{after - before});
}